packages/core/src/tracing/cls.js

```diff
@@ -155,238 +169,290 @@ class InstanaPseudoSpan extends InstanaSpan {
   }
 }
 
-/*
- * Start a new span and set it as the current span.
+/**
+ * @param {Function} fn
  */
-exports.startSpan = function startSpan(spanName, kind, traceId, parentSpanId, w3cTraceContext) {
-  tracingMetrics.incrementOpened();
-  if (!kind || (kind !== ENTRY && kind !== EXIT && kind !== INTERMEDIATE)) {
-    logger.warn('Invalid span (%s) without kind/with invalid kind: %s, assuming EXIT.', spanName, kind);
-    kind = EXIT;
-  }
-  const span = new InstanaSpan(spanName);
-  span.k = kind;
-
-  const parentSpan = exports.getCurrentSpan();
-  const parentW3cTraceContext = exports.getW3cTraceContext();
-
-  if (serviceName != null && !parentSpan) {
-    span.data.service = serviceName;
-  }
+function call(fn) {
+  fn();
+}
 
-  // If the client code has specified a trace ID/parent ID, use the provided IDs.
-  if (traceId) {
-    span.t = traceId;
-    if (parentSpanId) {
-      span.p = parentSpanId;
+module.exports = {
```

Comment:
@basti1302 
This is a "big" change for the cls module, so I am tagging you earlier, in case you disagree/have questions/suggestions.
Basically, this has no effect to whomever require the cls module.
The difference is that TS recognizes automagically all the type annotations of each exported method already, so I don't have to do the job twice.
Why twice you may ask? Because we are exporting method by method, but from the importers perspective, this is one module (cls) with lots of methods inside. In this case, TS does not understand (at least in the way we are doing this) that all methods belong to the whole cls module. This makes me:
 1. Annotate types for each individual method
 2. Annotate a "cls typedef" where I need to repeat the signature of each method and tell TS: hey this is the type that this module represents. 
 3. Then, because of item 2, for each module that requires cls, I need to say: hey use this typedef in order to know what is inside of cls

We can also, as always jump into a call to discuss

Reply:
one note: right now, I am running CI tests to get problems, and fixing them

Reply:
From my experience, it *eventually* happens that a code review becomes more complex to be done depending on the changes, which seems to be totally acceptable.
I would probably raise a concern too, if it was a recurring situation, instead of bringing that up in the very first time that such thing happens, but that's how I think, of course.
We can discuss in person which level of "pickiness" we want for our CR sessions so to prevent undesired code changes that one does not like to look at.

About the question, I think that it could be done in the way that you mentioned. I honestly cannot remember why I chose to do it in this particular way, since the PR is sitting there for quite some time.
My suspicion is that the work to do such thing should not be too big, so I will give it a try, and if I stumble into any unforeseen issue, I let you know.